examples/conversation_with_agent_with_finetuned_model/finetune_dialogagent.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+"""
+This module provides the Finetune_DialogAgent class,
+which extends DialogAgent to enhance fine-tuning
+capabilities with custom hyperparameters.
+"""
+from typing import Any, Optional, Dict
+
+from loguru import logger
+
+from agentscope.agents import DialogAgent
+
+
+class Finetune_DialogAgent(DialogAgent):
```

Comment:
The agent depends on `HuggingfaceModelWrapper`, should check the model type during initialization and raise appropriate error messages when the user attempts to load other models.

Reply:
> Rename to `FinetuneDialogAgent`.

Renamed FinetuneDialogAgent.

> The agent depends on `HuggingfaceModelWrapper`, should check the model type during initialization and raise appropriate error messages when the user attempts to load other models.

The current implementation can already handle such error. Example:
setting `"pretrained_model_name_or_path": 'Alireza1044/albert-base-v2-stsb',`
behavior:
```
2024-05-25 12:55:54.990 | INFO     | agentscope.models:read_model_configs:180 - Load configs for model wrapper: my_custom_model
2024-05-25 12:55:55.005 | INFO     | agentscope.utils.monitor:_create_monitor_table:396 - Init [monitor_metrics] as the monitor table
2024-05-25 12:55:55.006 | INFO     | agentscope.utils.monitor:_create_monitor_table:397 - Init [monitor_metrics_quota_exceeded] as the monitor trigger
2024-05-25 12:55:55.006 | INFO     | agentscope.utils.monitor:__init__:366 - SqliteMonitor initialization completed at [./runs/run_20240525-125554_e4eokx/agentscope.db]
2024-05-25 12:55:55.007 | INFO     | agentscope.models.model:__init__:201 - Initialize model by configuration [my_custom_model]
2024-05-25 12:55:55.338 | ERROR    | huggingface_model:load_model:284 - Failed to load model 'Alireza1044/albert-base-v2-stsb' from 'None': Unrecognized configuration class <class 'transformers.models.albert.configuration_albert.AlbertConfig'> for this kind of AutoModel: AutoModelForCausalLM.
Model type should be one of BartConfig, BertConfig, BertGenerationConfig, BigBirdConfig, BigBirdPegasusConfig, BioGptConfig, BlenderbotConfig, BlenderbotSmallConfig, BloomConfig, CamembertConfig, LlamaConfig, CodeGenConfig, CohereConfig, CpmAntConfig, CTRLConfig, Data2VecTextConfig, DbrxConfig, ElectraConfig, ErnieConfig, FalconConfig, FuyuConfig, GemmaConfig, GitConfig, GPT2Config, GPT2Config, GPTBigCodeConfig, GPTNeoConfig, GPTNeoXConfig, GPTNeoXJapaneseConfig, GPTJConfig, JambaConfig, JetMoeConfig, LlamaConfig, MambaConfig, MarianConfig, MBartConfig, MegaConfig, MegatronBertConfig, MistralConfig, MixtralConfig, MptConfig, MusicgenConfig, MusicgenMelodyConfig, MvpConfig, OlmoConfig, OpenLlamaConfig, OpenAIGPTConfig, OPTConfig, PegasusConfig, PersimmonConfig, PhiConfig, Phi3Config, PLBartConfig, ProphetNetConfig, QDQBertConfig, Qwen2Config, Qwen2MoeConfig, RecurrentGemmaConfig, ReformerConfig, RemBertConfig, RobertaConfig, RobertaPreLayerNormConfig, RoCBertConfig, RoFormerConfig, RwkvConfig, Speech2Text2Config, StableLmConfig, Starcoder2Config, TransfoXLConfig, TrOCRConfig, WhisperConfig, XGLMConfig, XLMConfig, XLMProphetNetConfig, XLMRobertaConfig, XLMRobertaXLConfig, XLNetConfig, XmodConfig.
Traceback (most recent call last):
  File "/home/zhan1130/agentscope/examples/conversation_with_agent_with_finetuned_model/conversation_with_agent_with_finetuned_model.py", line 114, in <module>
    main()
  File "/home/zhan1130/agentscope/examples/conversation_with_agent_with_finetuned_model/conversation_with_agent_with_finetuned_model.py", line 69, in main
    dialog_agent = FinetuneDialogAgent(
  File "/home/zhan1130/agentscope/src/agentscope/agents/agent.py", line 82, in __call__
    instance = super().__call__(*args, **kwargs)
  File "/home/zhan1130/agentscope/examples/conversation_with_agent_with_finetuned_model/FinetuneDialogAgent.py", line 49, in __init__
    super().__init__(
  File "/home/zhan1130/agentscope/src/agentscope/agents/dialog_agent.py", line 45, in __init__
    super().__init__(
  File "/home/zhan1130/agentscope/src/agentscope/agents/agent.py", line 195, in __init__
    self.model = load_model_by_config_name(model_config_name)
  File "/home/zhan1130/agentscope/src/agentscope/models/__init__.py", line 117, in load_model_by_config_name
    return _get_model_wrapper(model_type=model_type)(**kwargs)
  File "/home/zhan1130/agentscope/examples/conversation_with_agent_with_finetuned_model/huggingface_model.py", line 95, in __init__
    self.load_model(
  File "/home/zhan1130/agentscope/examples/conversation_with_agent_with_finetuned_model/huggingface_model.py", line 249, in load_model
    self.model = AutoModelForCausalLM.from_pretrained(
  File "/home/zhan1130/anaconda3/envs/agentscope/lib/python3.9/site-packages/transformers/models/auto/auto_factory.py", line 566, in from_pretrained
    raise ValueError(
ValueError: Unrecognized configuration class <class 'transformers.models.albert.configuration_albert.AlbertConfig'> for this kind of AutoModel: AutoModelForCausalLM.
Model type should be one of BartConfig, BertConfig, BertGenerationConfig, BigBirdConfig, BigBirdPegasusConfig, BioGptConfig, BlenderbotConfig, BlenderbotSmallConfig, BloomConfig, CamembertConfig, LlamaConfig, CodeGenConfig, CohereConfig, CpmAntConfig, CTRLConfig, Data2VecTextConfig, DbrxConfig, ElectraConfig, ErnieConfig, FalconConfig, FuyuConfig, GemmaConfig, GitConfig, GPT2Config, GPT2Config, GPTBigCodeConfig, GPTNeoConfig, GPTNeoXConfig, GPTNeoXJapaneseConfig, GPTJConfig, JambaConfig, JetMoeConfig, LlamaConfig, MambaConfig, MarianConfig, MBartConfig, MegaConfig, MegatronBertConfig, MistralConfig, MixtralConfig, MptConfig, MusicgenConfig, MusicgenMelodyConfig, MvpConfig, OlmoConfig, OpenLlamaConfig, OpenAIGPTConfig, OPTConfig, PegasusConfig, PersimmonConfig, PhiConfig, Phi3Config, PLBartConfig, ProphetNetConfig, QDQBertConfig, Qwen2Config, Qwen2MoeConfig, RecurrentGemmaConfig, ReformerConfig, RemBertConfig, RobertaConfig, RobertaPreLayerNormConfig, RoCBertConfig, RoFormerConfig, RwkvConfig, Speech2Text2Config, StableLmConfig, Starcoder2Config, TransfoXLConfig, TrOCRConfig, WhisperConfig, XGLMConfig, XLMConfig, XLMProphetNetConfig, XLMRobertaConfig, XLMRobertaXLConfig, XLNetConfig, XmodConfig.
```